scripts/init-mptcp.sh

```diff
@@ -0,0 +1,33 @@
+echo "Setting up MPTCP..."
+
+vagrant ssh client <<EOF
+    # Sometimes the ip addresses aren't assigned by virtualbox, so do it manually
+    sudo ip addr add 192.168.56.100/24 dev eth1
+    sudo ip addr add 192.168.57.100/24 dev eth2
+    sudo ip addr add 192.168.58.100/24 dev eth3
+
+    # Configure MPTCP
+    sudo sysctl net.mptcp.enabled=1
+    sudo ip mptcp limits set subflow 2
```

Comment:
why limiting to 2 subflows? Here, you have 3² possible paths.

Reply:
because any interface is connected to all interfaces of the peer ?

Reply:
OK, you cannot has crossed paths then? In this case, yes, that should work. Note that if you remove the first subflow and add a new one, it will be count as a new extra subflow. Maybe safer to set the limit to 3.

Reply:
Yes no crossed paths, only 3 virtual links, like in this picture ![](https://www.redhat.com/rhdc/managed-files/styles/wysiwyg_full_width/private/multipath-tcp-image1.png.webp?itok=F-3q2uFF)

I'll put then 3 sub flows, thanks for the explanations

Reply:
Update: after having checked, it seems that the network doesn't behave like 3 links, but instead like having crossed-paths (we can indeed ping the other interfaces eth2/eth3 of the server by using the interface eth1 of the client)